lib/sqlalchemy/ext/asyncio/session.py

```diff
@@ -1436,6 +1436,40 @@ async def main():
 
     class_: Type[_AS]
 
+    @overload
+    def __init__(
+        self: "async_sessionmaker[AsyncSession]",
+        bind: Optional[_AsyncSessionBind] = ...,
+        *,
+        autoflush: bool = ...,
+        expire_on_commit: bool = ...,
+        info: Optional[_InfoType] = ...,
+        **kw: Any,
+    ):
+        ...
+    @overload
+    def __init__(
+        self,
+        bind: Optional[_AsyncSessionBind],
+        class_: Type[_AS],
+        autoflush: bool = ...,
+        expire_on_commit: bool = ...,
+        info: Optional[_InfoType] = ...,
+        **kw: Any,
+    ):
+        ...
+    @overload
+    def __init__(
+        self,
+        bind: Optional[_AsyncSessionBind] = ...,
+        *,
+        class_: Type[_AS],
+        autoflush: bool = ...,
+        expire_on_commit: bool = ...,
+        info: Optional[_InfoType] = ...,
+        **kw: Any,
+    ):
```

Comment:
So, first overload: No class_ provided, self is explicitly `[AsyncSession]`.
Second overload: class_ and bind are provided.
Third overload: class_ is provided as a keyword value, bind is maybe not provided.